src/inmanta/server/agentmanager.py

```diff
@@ -168,12 +169,63 @@ def __init__(self, closesessionsonstart: bool = True, fact_back_off: Optional[in
 
         self.closesessionsonstart: bool = closesessionsonstart
 
-    async def get_status(self) -> dict[str, ArgumentTypes]:
-        return {
+    async def get_status(self) -> Mapping[str, ArgumentTypes | Mapping[str, ArgumentTypes]]:
```

Comment:
This method is very tricky, as it has to return withing the timeout of the status endpoint.

I chose to make it report what it has, even if that makes the format somewhat inconsistent. 
(e.g. if the DB is down, the environments will be reported by UUID instead of name)